src/server/ApiDispatchers.cpp

```diff
@@ -35,35 +47,21 @@
 {
     Telemetry::Instance().LogApiCall(Telemetry::ApiCall::GetConsoleMode);
     const auto a = &m->u.consoleMsgL1.GetConsoleMode;
-    std::wstring_view handleType = L"unknown";
-
-    TraceLoggingWrite(g_hConhostV2EventTraceProvider,
-                      "API_GetConsoleMode",
-                      TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
-                      TraceLoggingKeyword(TIL_KEYWORD_TRACE),
-                      TraceLoggingOpcode(WINEVENT_OPCODE_START));
-
-    auto tracing = wil::scope_exit([&]() {
-        Tracing::s_TraceApi(a, handleType);
-        TraceLoggingWrite(g_hConhostV2EventTraceProvider,
-                          "API_GetConsoleMode",
-                          TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
-                          TraceLoggingKeyword(TIL_KEYWORD_TRACE),
-                          TraceLoggingOpcode(WINEVENT_OPCODE_STOP));
-    });
 
     const auto pObjectHandle = m->GetObjectHandle();
     RETURN_HR_IF_NULL(E_HANDLE, pObjectHandle);
+
+    TraceConsoleAPICallWithOrigin("GetConsoleMode",
+                                  TraceLoggingBool(pObjectHandle->IsInputHandle(), "InputHandle"));

```

Comment:
nit: If you add a newline past `TraceConsoleAPICallWithOrigin(` it would consume less horizontal space, which might be useful when viewing text side-by-side. But it's fairly subjective if that's a visual improvement or not. 😅

Reply:
Ah, good call. I was wondering how to make it prettier!